core-libs/setup/ssr/optimized-engine/rendering-strategy-resolver.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Request } from 'express';
+import {
+  RenderingBlocked,
+  RenderingStrategy,
+} from './ssr-optimization-options';
+
+const hasExcludedParams = (
+  request: Request,
+  renderingBlocked: RenderingBlocked
+) => {
+  const params = request.query ? Object.getOwnPropertyNames(request.query) : [];
+
+  if (!renderingBlocked?.params) {
+    return false;
+  }
+
+  return renderingBlocked.params.some((excludedParam) =>
+    params.some((param) => excludedParam === param)
+  );
+};
+
+const hasExcludedUrl = (
+  request: Request,
+  renderingBlocked: RenderingBlocked
+) => {
+  return request.url && renderingBlocked?.urls
+    ? renderingBlocked.urls.some((url) => request.url.search(url) > -1)
+    : false;
+};
+
+const shouldFallbackToCsr = (
+  request: Request,
+  renderingBlocked: RenderingBlocked
```

Comment:
`could:` consider destructuring the object in this place and pass to `hasExcludedUrl` and `hasExcludedParams` only related values instead of the whole object.

Reply:
fixed